protocol/daemons/liquidation/client/sub_task_runner.go

```diff
@@ -82,39 +167,120 @@ func (c *Client) GetLiquidatableSubaccountIds(
+	liquidatableSubaccountIds = make([]satypes.SubaccountId, 0)
 	for _, subaccount := range subaccounts {
-		if len(subaccount.PerpetualPositions) > 0 {
-			subaccountsToCheck = append(subaccountsToCheck, *subaccount.Id)
+		// Skip subaccounts with no open positions.
+		if len(subaccount.PerpetualPositions) == 0 {
+			numSubaccountsWithOpenPositions++
+			continue
+		}
+
+		// Check if the subaccount is liquidatable.
+		isLiquidatable, err := c.CheckSubaccountCollateralization(
+			subaccount,
+			marketPrices,
+			perpetuals,
+			liquidityTiers,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		if isLiquidatable {
+			liquidatableSubaccountIds = append(liquidatableSubaccountIds, *subaccount.Id)
 		}
 	}
 
 	telemetry.ModuleSetGauge(
 		metrics.LiquidationDaemon,
-		float32(len(subaccountsToCheck)),
+		float32(numSubaccountsWithOpenPositions),
 		metrics.SubaccountsWithOpenPositions,
 		metrics.Count,
 	)
 
-	// Query the gRPC server in chunks of size `liqFlags.RequestChunkSize`.
-	liquidatableSubaccountIds = make([]satypes.SubaccountId, 0)
-	for start := 0; start < len(subaccountsToCheck); start += int(liqFlags.RequestChunkSize) {
-		end := lib.Min(start+int(liqFlags.RequestChunkSize), len(subaccountsToCheck))
+	return liquidatableSubaccountIds, nil
+}
 
-		results, err := c.CheckCollateralizationForSubaccounts(
-			ctx,
-			subaccountsToCheck[start:end],
-		)
-		if err != nil {
-			return nil, err
+// CheckSubaccountCollateralization queries a gRPC server using `AreSubaccountsLiquidatable`
+// and returns a list of collateralization statuses for the given list of subaccount ids.
+func (c *Client) CheckSubaccountCollateralization(
+	subaccount satypes.Subaccount,
+	marketPrices map[uint32]pricestypes.MarketPrice,
+	perpetuals map[uint32]perptypes.Perpetual,
+	liquidityTiers map[uint32]perptypes.LiquidityTier,
+) (
+	isLiquidatable bool,
+	err error,
+) {
+	defer telemetry.ModuleMeasureSince(
+		metrics.LiquidationDaemon,
+		time.Now(),
+		metrics.CheckCollateralizationForSubaccounts,
+		metrics.Latency,
+	)
+
+	bigTotalNetCollateral := big.NewInt(0)
+	bigTotalMaintenanceMargin := big.NewInt(0)
+
+	// Calculate the net collateral and maintenance margin for each of the asset positions.
+	// Note that we only expect USDC before multi-collateral support is added.
+	for _, assetPosition := range subaccount.AssetPositions {
+		if assetPosition.AssetId != assetstypes.AssetUsdc.Id {
+			panic("liquidation daemon only supports USDC collateral")
+		}
+		// Net collateral for USDC is the quantums of the position.
+		// Margin requirements for USDC are zero.
+		bigTotalNetCollateral.Add(bigTotalNetCollateral, assetPosition.GetBigQuantums())
+	}
+
+	// Calculate the net collateral and maintenance margin for each of the perpetual positions.
+	for _, perpetualPosition := range subaccount.PerpetualPositions {
+		perpetual, ok := perpetuals[perpetualPosition.PerpetualId]
+		if !ok {
+			panic(
+				fmt.Sprintf(
+					"Perpetual not found for perpetual id %d",
```

Comment:
Can we define errors for these and panic with an `error` instead of a string? I believe it would work better with the dd error tracking when we recover the panic, we will get better error tracking.

Reply:
actually I updated to return error instead. caller will logger.Error and propagate error to upstream